docs/src/main/sphinx/connector/hive.rst

```diff
@@ -537,63 +578,68 @@ In order to use a Glue catalog, ensure to configure the metastore with
-                                                     Example: ``us-east-1``
+======================================================= ============================================================
+Property Name                                           Description
+======================================================= ============================================================
+``hive.metastore.glue.region``                          AWS region of the Glue Catalog. This is required when not
+                                                        running in EC2, or when the catalog is in a different region.
+                                                        Example: ``us-east-1``
+
+``hive.metastore.glue.endpoint-url``                    Glue API endpoint URL (optional).
+                                                        Example: ``https://glue.us-east-1.amazonaws.com``
 
-``hive.metastore.glue.endpoint-url``                 Glue API endpoint URL (optional).
-                                                     Example: ``https://glue.us-east-1.amazonaws.com``
+``hive.metastore.glue.pin-client-to-current-region``    Pin Glue requests to the same region as the EC2 instance
+                                                        where Trino is running, defaults to ``false``.
 
-``hive.metastore.glue.pin-client-to-current-region`` Pin Glue requests to the same region as the EC2 instance
-                                                     where Trino is running, defaults to ``false``.
+``hive.metastore.glue.max-connections``                 Max number of concurrent connections to Glue,
+                                                        defaults to ``5``.
 
-``hive.metastore.glue.max-connections``              Max number of concurrent connections to Glue,
-                                                     defaults to ``5``.
+``hive.metastore.glue.max-error-retries``               Maximum number of error retries for the Glue client,
+                                                        defaults to ``10``.
 
-``hive.metastore.glue.max-error-retries``            Maximum number of error retries for the Glue client,
-                                                     defaults to ``10``.
+``hive.metastore.glue.default-warehouse-dir``           Default warehouse directory for schemas created without an
+                                                        explicit ``location`` property.
 
-``hive.metastore.glue.default-warehouse-dir``        Default warehouse directory for schemas created without an
-                                                     explicit ``location`` property.
+``hive.metastore.glue.aws-credentials-provider``        Fully qualified name of the Java class to use for obtaining
+                                                        AWS credentials. Can be used to supply a custom credentials
```

Comment:
Related question not to be answered here...

how to you add access to that class.. drop the related jars into the Hive connector plugin folder? 

If this is a common use case we might need to document specifically .. or generically somewhere

Reply:
https://github.com/trinodb/trino/blob/f7e0a0d345086cf59ab69c0e1477c500a267b3c5/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueCredentialsProvider.java#L66-L78

I'm assuming that this setting was thought to deal with one of the many credential providers which come with the `aws-java-sdk-core` library.

If the class is not in the library `aws-java-sdk-core` then it would need to be probably added to hive's plugin folder.

Related PR: https://github.com/trinodb/trino/pull/1363